lib/spack/spack/hooks/__init__.py

```diff
@@ -28,71 +28,61 @@
 systems (e.g. modules, lmod, etc.) or to add other custom
 features.
 """
-import llnl.util.lang
+import contextlib
 
-import spack.paths
 
-
-class _HookRunner(object):
+class HookRunner(object):
     #: Stores all hooks on first call, shared among
     #: all HookRunner objects
     _hooks = None
 
-    def __init__(self, hook_name):
-        self.hook_name = hook_name
+    def __init__(self, hooks=[]):
```

Comment:
It's better not to have lists as default arguments, since they are mutable objects created at function definition time and reused for each function call. You can set the default to `None` and:
```python
self.hooks_name = hooks or []
```